plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -143,14 +143,41 @@ fu_plugin_flashrom_device_set_hwids(FuPlugin *plugin, FuDevice *device)
 }
 
 static gboolean
-fu_plugin_flashrom_coldplug(FuPlugin *plugin, GError **error)
+fu_plugin_flashrom_add_device(FuPlugin *plugin, const gchar *guid, FuIfdRegion region,
+			      GError **error)
 {
 	FuContext *ctx = fu_plugin_get_context(plugin);
 	const gchar *dmi_vendor;
-	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx);
+	const gchar *product = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_PRODUCT_NAME);
+	const gchar *vendor = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER);
+	const gchar *region_str = fu_ifd_region_to_string(region);
+	g_autofree gchar *name = g_strdup_printf("%s (%s)", product, region_str);
+	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx, region);
+
+	fu_device_set_name(device, name);
+	fu_device_set_vendor(device, vendor);
+
+	fu_device_add_instance_str(device, "REGION", region_str);
+	fu_device_add_instance_str(device, "VENDOR", vendor);
+	fu_device_add_instance_str(device, "PRODUCT", product);
+	if (!fu_device_build_instance_id(device, NULL, "FLASHROM", "VENDOR", "PRODUCT", "REGION",
+					 NULL)) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "flashrom failed to add instance id for %s device", region_str);
+		return FALSE;
+	}
 
-	fu_device_set_name(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_PRODUCT_NAME));
-	fu_device_set_vendor(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER));
+	/* add this so we can attach all the other quirks */
+	fu_device_add_instance_str(FU_DEVICE(device), "GUID", guid);
+	if (!fu_device_build_instance_id(FU_DEVICE(device), error, "FLASHROM", "GUID", NULL)) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "flashrom failed to add GUID instance id for %s device", region_str);
```

Comment:
you don't need this, fu_device_build_instance_id will set the error for you.

Reply:
Sure, thanks, varargs confused me.